docs/user-guide/rules/http-compression.md

```diff
@@ -0,0 +1,567 @@
+
+However, there are a few things that need to be done right in order for
+get the most out of compression:
+
+* Only compress resources for which the result of the compression
+  will be smaller than original size.
+
+  In general text-based resources (HTML, CSS, JavaScript, SVGs, etc.)
+  compresss very well especially if the file is not very small.
+  The same goes for some other file formats (e.g.: ICO files, web fonts
+  such as EOT, OTF, and TTF, etc.)
+
+  However, compressing resources that are already compressed (e.g.:
+  images, audio files, PDFs, etc.) not only waste CPU resources, but
+  usually result in little to no reduction, or in some cases even
+  a bigger file size.
+
+  The same goes for resources that are very small because of the
+  overhead of compression file formats.
+
+* Use most efficien compression method.
+
+  `gzip` is the most used encoding nowadays as it strikes a good
+  balance between compression ratio (as high as 70-90% especially
+  for larger files) and encoding time, and is supported pretty much
+  eveywere.
+
+  Better savings can be achieved using [`Zopfli`][zopfli] which
+  can reduce the size on average with [3–8% more then `gzip`][[zopfli
+  blog post]. Since `Zopfli` output (for the `gzip` option) is valid
+  `gzip` content, `Zopfli` works eveywere `gzip` works. The only
+  downsize is that encoding takes way more time than with `gzip`,
+  thus, making `Zopfli` more suitable for static content (i.e. encoding
+  resources with is as part of build script, not on the fly).
+
+  But, things can be improved even futher using [Brotli][brotli]. This
+  new encoding allows to get [20–26% higher compression ratios][brotli
+  blog post] even over `Zopfli`. However, being a new encoding and
+  that is not compatible with `gzip`, limits its support to modern
+  browsers, and its usage to [only over HTTPS (as proxies misinterpreting
+  unknown encodings)][brotli over https].
+
+  So, in general, for best performance and interoperability resources
+  should be served compress with `Zopfli` over HTTP and `Brotli` over
+  HTTPS.
+
+* Avoid using deprecated or not widlly supported compression formats,
+  and `Content-Type` values.
+
+  Avoid using deprecated `Content-Type` values such as `x-gzip`. Some
+  user agents may alias them to the correct, current equivalent value
+  (e.g.: alias `x-gzip` to `gzip`), but that is not always true.
+
+  Also avoid using encoding that are not widely supported (e.g.:
+  `compress`, `bzip2`, [`SDCH`][unship sdch], etc.), and/or may not
+  be as efficient, or can create problems (e.g.: [`deflate`][deflate
+  issues]). In general these should be avoided, and one should just
+  stick to the encoding specified in the previous point.
+
+* Avoid potential caching related issues.
+
+  When resources are served compressed, they should be served with
+  the `Vary` header containing the `Accept-Encoding` value (or with
+  something such as `Cache-Control: private` that prevents caching
```

Comment:
> that prevents caching in proxy caches and such altogether
 
The final user can cache the response so we are not preventing caching altogether:

> Indicates that all or part of the response message is intended for a single user and MUST NOT be cached by a shared cache, such as a proxy server.

[source](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html)

Reply:
Fixed.